test/Grpc.Net.Client.Tests/CallCredentialTests.cs

```diff
@@ -79,19 +80,67 @@ public async Task CallCredentialsWithHttps_MetadataOnRequest()
         var invoker = HttpClientCallInvokerFactory.Create(httpClient);
 
         // Act
+        var tcs = new TaskCompletionSource<object?>(TaskCreationOptions.RunContinuationsAsynchronously);
         var callCredentials = CallCredentials.FromInterceptor(async (context, metadata) =>
         {
             // The operation is asynchronous to ensure delegate is awaited
-            await Task.Delay(50);
+
+            // Ensure task hasn't been completed.
+            Assert.False(tcs.Task.IsCompleted);
```

Comment:
This is a bit odd. I think in theory the compiler or synccontext could choose to run this asynchronously before it gets here in which case it would be a race.

Generally you would instead use 2 tcs's to ensure order.

Reply:
I _think_ the test was fine. The tcs couldn't be completed until after the auth interceptor awaits. The assert was to verify that behavior.

But I agree the test looks a little weird. And it's brittle. I've switched it to use a sync point.